src/gui/layout/qgslayoutlegendwidget.cpp

```diff
@@ -1166,29 +1166,17 @@ void QgsLayoutLegendWidget::mLayerExpressionButton_clicked()
   QgsExpressionContext legendContext = mLegend->createExpressionContext();
   legendContext.appendScope( vl->createExpressionContextScope() );
 
-  QgsExpressionContextScope *symbolLegendScope = new QgsExpressionContextScope( tr( "Symbol scope" ) );
-
-  QgsFeatureRenderer *r = vl->renderer();
-
   QStringList highlighted;
-  if ( r )
+  const QList<QgsLayerTreeModelLegendNode *> legendnodes = mLegend->model()->layerLegendNodes( layerNode, false );
```

Comment:
is `mLegend` could be a nullptr? same for `model()`?

Reply:
`mLegend` being nullptr is handled at the beginning of the method.

`model()` can't be nullptr, it is set in the object constructor, and no method that are using it are testing it. But I guess it doesn't harm to test here :)